applications/app/controllers/InteractiveController.scala

```diff
@@ -68,121 +64,90 @@ class InteractiveController(
+    val dataJson = DotcomRenderingDataModel.toJson(data)
+    val res = common.renderJson(dataJson, page).as("application/json")
+    Future.successful(res)
   }
 
-  def renderDCRAmp(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) flatMap {
-      case Left((model, blocks)) => {
-        val pageType = PageType.apply(model, request, context)
-        remoteRenderer.getAMPInteractive(wsClient, model, blocks, pageType)
-      }
-      case Right(other) => Future.successful(RenderOtherStatus(other))
-    }
-  }
-
-  def itemResponseToInteractivePickerInputData(item: ItemResponse)(implicit
-      request: RequestHeader,
-  ): Option[(DateTime, List[Tag])] = {
-    for {
-      capiDatetime <- item.content.flatMap(_.webPublicationDate)
-      datetime = DateTime.parse(capiDatetime.iso8601)
-      tags <- item.content.map(_.tags)
-    } yield (datetime, tags.toList)
+  def renderAmp(page: InteractivePage, blocks: Blocks)(implicit request: RequestHeader): Future[Result] = {
+    val pageType = PageType.apply(page, request, context)
+    remoteRenderer.getAMPInteractive(wsClient, page, blocks, pageType)
   }
 
   override def renderItem(path: String)(implicit request: RequestHeader): Future[Result] = {
     val requestFormat = request.getRequestFormat
-    /*
-       Calling lookupItemResponse(path) on an election tracker path, 404 in CAPI, the atom is retrived using the atom Id.
-       This is the reason why we need to process those paths before performing the logic for the Interactive Picker
-       (ie: before calling the CAPI response and extracting the input parameters for `getRenderingTier`)
-     */
-    if (USElection2020AmpPages.pathIsSpecialHanding(path) && requestFormat == AmpFormat) {
-      renderInteractivePageUSPresidentialElection2020(path)
-    } else {
-      val res = lookupItemResponse(path).flatMap { itemResponse =>
-        itemResponseToInteractivePickerInputData(itemResponse) match {
-          case Some((datetime, tags)) => {
-            val renderingTier = InteractivePicker.getRenderingTier(path, datetime, tags)
-            (requestFormat, renderingTier) match {
-              case (AmpFormat, DotcomRendering)  => renderDCRAmp(itemResponse)
-              case (JsonFormat, DotcomRendering) => renderDCRJson(itemResponse)
-              case (HtmlFormat, DotcomRendering) => renderDCR(itemResponse)
-              case _                             => renderItemLegacy(itemResponse)
-            }
+    val isUSElectionAMP = USElection2020AmpPages.pathIsSpecialHanding(path) && requestFormat == AmpFormat
+
+    def render(model: Either[(InteractivePage, Blocks), Result]): Future[Result] = {
+      model match {
+        case Left((page, blocks)) => {
+          val tags = page.interactive.tags.tags
+          val date = page.interactive.trail.webPublicationDate
```

Comment:
We now grab the date from the model, rather than CAPI response. It turns out that the model code will use `now` for the date here if there was none on the CAPI response. I think this occurs for unpublished content only (so actually what we want to correct preview for DCR) but if anyone knows better please shout! cc. @shtukas who might have thoughts here.